File: fixtures/scale/ruby/large/lib/check_stairstep.rb
require_relative 'stairstep_encrypt'
require_relative 'stairstep_decrypt'
require_relative 'corpus_additive'

def check_stairstep
  corpus_additive.each do |text|
    sealed = stairstep_encrypt(text, 7)
    opened = stairstep_decrypt(sealed, 7)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_stairstep.py
from cipherlab.ciphers.stairstep import stairstep_encrypt, stairstep_decrypt
from cipherlab.specs.stairstep_spec import stairstep_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_stairstep():
    spec = stairstep_spec()
    for text in corpus_additive():
        sealed = stairstep_encrypt(text, spec["key"])
        opened = stairstep_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
